src/os/data/columndefinition.js

```diff
@@ -221,30 +254,62 @@ os.data.ColumnDefinition.prototype.restore = function(config) {
   this['id'] = config['id'];
   this['name'] = config['name'];
   this['field'] = config['field'];
-  this['type'] = config['type'] || 'string';
-  this['resizable'] = config['resizable'];
-  this['sortable'] = config['sortable'];
-  this['minWidth'] = config['minWidth'];
-  this['maxWidth'] = config['maxWidth'];
-  this['width'] = config['width'];
-  this['rerenderOnResize'] = config['rerenderOnResize'];
-  this['headerCssClass'] = config['headerCssClass'];
-  this['cssClass'] = config['cssClass'];
-  this['defaultSortAsc'] = config['defaultSortAsc'];
-  this['focusable'] = config['focusable'];
-  this['selectable'] = config['selectable'];
-  this['behavior'] = config['behavior'];
-  this['visible'] = config['visible'];
-  this['visprotected'] = config['visprotected'];
-  this['toolTip'] = config['toolTip'];
-  this['derivedFrom'] = config['derivedFrom'];
-  this['userModified'] = config['userModified'];
 
-  /**
-   * @todo Need a way to persist/restore these.
-   */
-  this['formatter'] = null;
-  this['asyncPostRender'] = null;
+  // only set these values if these keys are on the config
+  if (config.hasOwnProperty('type')) {
+    this['type'] = config['type'];
+  }
+  if (config.hasOwnProperty('resizable')) {
+    this['resizable'] = config['resizable'];
+  }
+  if (config.hasOwnProperty('sortable')) {
+    this['sortable'] = config['sortable'];
+  }
+  if (config.hasOwnProperty('minWidth')) {
+    this['minWidth'] = config['minWidth'];
+  }
+  if (config.hasOwnProperty('maxWidth')) {
+    this['maxWidth'] = config['maxWidth'];
+  }
+  if (config.hasOwnProperty('width')) {
+    this['width'] = config['width'];
+  }
+  if (config.hasOwnProperty('rerenderOnResize')) {
+    this['rerenderOnResize'] = config['rerenderOnResize'];
+  }
+  if (config.hasOwnProperty('headerCssClass')) {
+    this['headerCssClass'] = config['headerCssClass'];
+  }
+  if (config.hasOwnProperty('cssClass')) {
+    this['cssClass'] = config['cssClass'];
+  }
+  if (config.hasOwnProperty('defaultSortAsc')) {
+    this['defaultSortAsc'] = config['defaultSortAsc'];
+  }
+  if (config.hasOwnProperty('focusable')) {
+    this['focusable'] = config['focusable'];
+  }
+  if (config.hasOwnProperty('selectable')) {
+    this['selectable'] = config['selectable'];
+  }
+  if (config.hasOwnProperty('behavior')) {
+    this['behavior'] = config['behavior'];
+  }
+  if (config.hasOwnProperty('visible')) {
+    this['visible'] = config['visible'];
+  }
+  if (config.hasOwnProperty('visprotected')) {
+    this['visprotected'] = config['visprotected'];
+  }
+  if (config.hasOwnProperty('toolTip')) {
+    this['toolTip'] = config['toolTip'];
+  }
+  if (config.hasOwnProperty('derivedFrom')) {
+    this['derivedFrom'] = config['derivedFrom'];
+  }
+  if (config.hasOwnProperty('userModified')) {
+    this['userModified'] = config['userModified'];
+  }
```

Comment:
This works, but it might be worthwhile for maintainability here to create a new "default" instance and loop over all the fields and compare them to the default. Then you do not have the default values listed in two places.

Reply:
Good call, I'll switch to something like that.